datadog/resource_datadog_security_monitoring_default_rule.go

```diff
@@ -197,6 +231,24 @@ func buildSecMonDefaultRuleUpdatePayload(currentState datadogV2.SecurityMonitori
 		payload.Cases = &updatedRuleCase
 	}
 
+	tfFilters := d.Get("filter").([]interface{})
```

Comment:
Nit: I think we can be more concrete in the type assertion here, which would save a further assertion further down (L240).
```suggestion
	tfFilters := d.Get("filter").([]map[string]interface{})
```

Reply:
😿 
```
panic: interface conversion: interface {} is []interface {}, not []map[string]interface {}
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog:
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: goroutine 102 [running]:
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/terraform-providers/terraform-provider-datadog/datadog.buildSecMonDefaultRuleUpdatePayload(0xc0006fc240, 0xc0006f42c0, 0xc0006f40e0, 0xc0006fc080, 0xc0006b6c20, 0xc0006f4590, 0xc0006f4455, 0xc0006f40e8, 0xc0006b6940, 0xc0006b6c30, ...)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/src/github.com/DataDog/terraform-provider-datadog/datadog/resource_datadog_security_monitoring_default_rule.go:231 +0xdc5
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogSecurityMonitoringDefaultRuleUpdate(0xc00028a8c0, 0x1fd0b20, 0xc000b26140, 0x24, 0x2f83fe0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/src/github.com/DataDog/terraform-provider-datadog/datadog/resource_datadog_security_monitoring_default_rule.go:155 +0x345
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000203170, 0xc0007b7400, 0xc0007d6f00, 0x1fd0b20, 0xc000b26140, 0x20bff01, 0xc0005b7c98, 0xc0007e40c0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/github.com/therve/terraform-plugin-sdk@v1.16.1-0.20210202202613-4d59f03d3b5f/helper/schema/resource.go:316 +0x273
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000310500, 0xc000d63a38, 0xc0007b7400, 0xc0007d6f00, 0xc0007dc628, 0xc000d93340, 0x20c2420)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/github.com/therve/terraform-plugin-sdk@v1.16.1-0.20210202202613-4d59f03d3b5f/helper/schema/provider.go:294 +0x99
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0001225a0, 0x2571240, 0xc0007ceba0, 0xc0004d1570, 0xc0001225a0, 0xc0007ceba0, 0xc0007b1ba0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/github.com/therve/terraform-plugin-sdk@v1.16.1-0.20210202202613-4d59f03d3b5f/internal/helper/plugin/grpc_provider.go:885 +0x8ab
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x2249c20, 0xc0001225a0, 0x2571240, 0xc0007ceba0, 0xc000b2f020, 0x0, 0x2571240, 0xc0007ceba0, 0xc000cf41e0, 0xf0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/github.com/therve/terraform-plugin-sdk@v1.16.1-0.20210202202613-4d59f03d3b5f/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003c3880, 0x257c480, 0xc0000c8f00, 0xc0007bc200, 0xc00021d7d0, 0x2f40a20, 0x0, 0x0, 0x0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:1171 +0x522
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: google.golang.org/grpc.(*Server).handleStream(0xc0003c3880, 0x257c480, 0xc0000c8f00, 0xc0007bc200, 0x0)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:1494 +0xcc5
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00030e360, 0xc0003c3880, 0x257c480, 0xc0000c8f00, 0xc0007bc200)
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:834 +0xa5
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-03-25T11:36:02.571+0100 [DEBUG] plugin.terraform-provider-datadog: 	/Users/alexej/go/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:832 +0x1fd
2021-03-25T11:36:02.575+0100 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-03-25T11:36:02.575+0100 [DEBUG] plugin: plugin process exited: path=/Users/alexej/go/bin/terraform-provider-datadog pid=79090 error="exit status 2"
2021/03/25 11:36:02 [DEBUG] datadog_security_monitoring_default_rule.test: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021/03/25 11:36:02 [TRACE] EvalWriteState: recording 0 dependencies for datadog_security_monitoring_default_rule.test
2021/03/25 11:36:02 [TRACE] EvalWriteState: writing current state object for datadog_security_monitoring_default_rule.test
2021/03/25 11:36:02 [TRACE] EvalApplyProvisioners: datadog_security_monitoring_default_rule.test is not freshly-created, so no provisioning is required
2021/03/25 11:36:02 [TRACE] EvalWriteState: recording 0 dependencies for datadog_security_monitoring_default_rule.test
2021/03/25 11:36:02 [TRACE] EvalWriteState: writing current state object for datadog_security_monitoring_default_rule.test
2021/03/25 11:36:02 [TRACE] vertex "datadog_security_monitoring_default_rule.test": visit complete
2021/03/25 11:36:02 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021/03/25 11:36:02 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/datadog/datadog\"] (close)" errored, so skipping
2021/03/25 11:36:02 [TRACE] dag/walk: upstream of "root" errored, so skipping
2021/03/25 11:36:02 [TRACE] statemgr.Filesystem: creating backup snapshot at terraform.tfstate.backup
2021/03/25 11:36:02 [TRACE] statemgr.Filesystem: state has changed since last snapshot, so incrementing serial to 127
2021/03/25 11:36:02 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate
2021/03/25 11:36:02 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2021/03/25 11:36:02 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2021-03-25T11:36:02.602+0100 [DEBUG] plugin: plugin exited

```

Reply:
unfortunately I had to keep it 
```
	tfFilters := d.Get("filter").([]interface{})
	payloadFilters := make([]datadogV2.SecurityMonitoringFilter, len(tfFilters))

	for idx, tfRuleFilter := range tfFilters {
		structRuleFilter := datadogV2.SecurityMonitoringFilter{}

		ruleFilter := tfRuleFilter.(map[string]interface{})
```
to avoid it panicking

Reply:
Ah, cool, thanks. 🙇 I'm happy with the PR, so it's up to the code owners to give it one last pass. 👍